src/error_logger_lager_h.erl

```diff
@@ -83,8 +83,37 @@ handle_call({set_high_water, N}, #state{shaper=Shaper} = State) ->
 handle_call(_Request, State) ->
     {ok, unknown_call, State}.
 
+shaper_fun() ->
+    case {lager_app:get_env(lager, suppress_supervisor_start_stop, false), lager_app:get_env(lager, suppress_application_start_stop, false)} of
+        {false, false} ->
+            fun(_) -> false end;
+        {true, true} ->
+            fun({info_report, _GL, {_Pid, std_info, D}}) when is_list(D) ->
+                    lists:member({exited, stopped}, D);
```

Comment:
I would really like to see this anonymous fun in its own named function - I think it will make the code clearer and any stacktraces more obvious.